src/Separated.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * ```ts
+ * interface Separated<A, B> {
+ *    readonly left: A
+ *    readonly right: B
+ * }
+ * ```
+ *
+ * Represents a result of separating a whole into two parts.
+ *
+ * @since 2.10.0
+ */
+
+import { Functor2 } from './Functor'
+import { Bifunctor2 } from './Bifunctor'
+
+// -------------------------------------------------------------------------------------
+// model
+// -------------------------------------------------------------------------------------
+
+/**
+ * A `Separated` type which holds `left` and `right` parts.
+ *
+ * @since 2.0.0
+ */
+export interface Separated<A, B> {
+  readonly left: A
+  readonly right: B
+}
+
+// -------------------------------------------------------------------------------------
+// constructors
+// -------------------------------------------------------------------------------------
+
+/**
+ * @category constructors
+ * @since 2.10.0
+ */
+export const separated = <A, B>(left: A, right: B): Separated<A, B> => ({ left, right })
+
+// -------------------------------------------------------------------------------------
+// non-pipeables
+// -------------------------------------------------------------------------------------
+
+const map_ = <A, B, C>(fa: Separated<A, B>, f: (b: B) => C): Separated<A, C> => ({ ...fa, right: f(fa.right) })
```

Comment:
I'd derive `map_` (`mapLeft_`, `bimap_`) from the corresponding pipeable function

```ts
const map_: Functor2<URI>['map'] = (fa, f) => pipe(fa, map(f))
```